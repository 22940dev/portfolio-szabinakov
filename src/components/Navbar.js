import React from 'react'
import { Link } from 'react-router-dom'
import '../styles/Navbar.css'

const Navbar = () => {

    return (
            <div data-testid='menuWrap' className='menuWrap'>
                <input data-testid='toggler' type='checkbox' className='toggler'></input>
                <div data-testid='hamburger' className='hamburger'>
                    <div></div>
                </div>
                <div data-testid='menu' className='menu'>
                    <div>
                        <div>
                            <ul>
                                <li>
                                <Link data-testid='navbarListElementHome' className='navbarListElementHome' to='/'>
                                    Home</Link>
                                </li>
                                <li>
                                <Link data-testid='navbarListElementProjects' className='navbarListElementProjects' to='/projects'>
                                    Projects</Link>
                                </li>
                                {/* <li>
                                <Link data-testid='navbarListElementContact' className='navbarListElementContact' to='/contact'>
                                    Contact</Link>
                                </li> */}
                            </ul>
                        </div>
                    </div>
                </div>
            </div>  
    )
}

export default Navbar