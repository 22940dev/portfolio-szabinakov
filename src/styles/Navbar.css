.menuWrap {
    position: fixed;
    top:0;
    left: 0;
    z-index: 1;
}
.menuWrap .toggler {
    position: absolute;
    top:0;
    left: 0;
    z-index: 2;
    cursor: pointer;
    width: 50px;
    height: 50px;
    opacity: 0;
}

.menuWrap .hamburger {
    position: absolute;
    top:0;
    left:0;
    z-index: 1;
    width: 30px;
    height: 30px;
    padding: 10px;
    display: flex;
    align-items: center;
    justify-content: center;
}

/* Hamburger Lines */

.menuWrap .hamburger > div {
    position: relative;
    width: 100%;
    height: 3px;
    border-radius: 30%;
    background-color: black;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 0.4s ease;
}

/* Top and Bottom lines */
.menuWrap .hamburger > div:before,
.menuWrap .hamburger > div:after {
    content: '';
    position: absolute;
    z-index: 1;
    top: -10px;
    width: 100%;
    height: 3px;
    border-radius: 30%;
    background-color: black;
}

/* Moves one line to the bottom */
.menuWrap .hamburger > div:after {
    top: 10px;
}

/* Toggler Animate */

.menuWrap .toggler:checked + .hamburger > div {
    transform: rotate(135deg);
}

/* Turn lines into X */
.menuWrap .toggler:checked + .hamburger > div:before,
.menuWrap .toggler:checked + .hamburger > div:after {
    top:0;
    transform: rotate(90deg);
}

/* Rotate on hover when checked */ 

.menuWrap .toggler:checked:hover + .hamburger > div {
    transform: rotate(225deg);
}

/* Show Menu */

.menuWrap .toggler:checked ~ .menu {
    visibility: visible;
}

.menuWrap .toggler:checked ~ .menu > div {
    transform: scale(1);
    transition-duration: 0.75s;
}

.menuWrap .toggler:checked ~ .menu > div > div {
    opacity: 1;
    transition: opacity 0.4s ease;
}

.menuWrap .menu {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    visibility: hidden; 
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
}

.menuWrap .menu > div {
    width: 100%;
    height: 50px;
    display: flex;
    flex: none;
    align-items: center;
    justify-content: center;
    transform: scale(0);
    transition: all 0.4s ease;
    border-bottom: solid 1px white;
}

.menuWrap .menu > div > div {
    text-align: center;
    max-width: 90vw;
    max-height: 100vh;
    opacity: 0;
    transition: opacity 0.4s ease;
}

.menuWrap .menu > div > div > ul {
    display: flex;
}

.menuWrap .menu > div > div > ul > li {
    list-style: none;
    color: black;
    font-size: 16px;
    padding: 10px;
}

.menuWrap .menu > div > div > ul > li > .navbarListElement {    
    color: inherit;
    text-decoration: none;
    transition: color 0.4s ease;
}


