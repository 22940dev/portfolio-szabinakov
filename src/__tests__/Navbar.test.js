import React from 'react'
import {render} from '@testing-library/react'
import Navbar from '../components/Navbar'
import { BrowserRouter } from 'react-router-dom'

describe('Navbar', () => {
    it('matches the snapshot', () => {
        const {asFragment} = render (
        <BrowserRouter>
            <Navbar/>
        </BrowserRouter>)

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders all elements' , () => {
        const {getByTestId} = render (
            <BrowserRouter>
            <Navbar/>
            </BrowserRouter>
        )

        expect(getByTestId('menuWrap')).toHaveClass('menuWrap')
        expect(getByTestId('hamburger')).toHaveClass('hamburger')
        expect(getByTestId('toggler')).toHaveClass('toggler')
        expect(getByTestId('menu')).toHaveClass('menu')
        expect(getByTestId('navbarListElementHome')).toHaveClass('navbarListElement')
        expect(getByTestId('navbarListElementContact')).toHaveClass('navbarListElement')
    })
})