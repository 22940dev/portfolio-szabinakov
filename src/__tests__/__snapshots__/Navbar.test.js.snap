// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="menuWrap"
    data-testid="menuWrap"
  >
    <input
      class="toggler"
      data-testid="toggler"
      type="checkbox"
    />
    <div
      class="hamburger"
      data-testid="hamburger"
    >
      <div />
    </div>
    <div
      class="menu"
      data-testid="menu"
    >
      <div>
        <div>
          <ul>
            <li>
              <a
                class="navbarListElement"
                data-testid="navbarListElementHome"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class="navbarListElement"
                data-testid="navbarListElementContact"
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
